From 5b6acdc3d665d4a03aff59be2f8160bef41c50b8 Mon Sep 17 00:00:00 2001
From: Yong Fu <fuyong@neusoft.com>
Date: Thu, 16 Jul 2020 16:36:29 +0800
Subject: [PATCH 2/3] modify file path and add dependencies

No permission to create a folder in the root directory causes the build to fail.
Therefore, the build directory needs to be changed.

Signed-off-by: Yong Fu <fuyong@neusoft.com>
---
 SPECS/docker-ce.spec      |  21 ++++--
 1 files changed, 21 insertions(+), 7 deletions(-)

diff --git a/SPECS/docker-ce.spec b/SPECS/docker-ce.spec
index f848122..ac1898a 100644
--- a/SPECS/docker-ce.spec
+++ b/SPECS/docker-ce.spec
@@ -19,6 +19,9 @@ URL: https://www.docker.com
 Vendor: Docker
 Packager: Docker <support@docker.com>
 
+# STX
+Patch0001:  modify-prefix.patch
+
 Requires: docker-ce-cli
 Requires: container-selinux >= 2:2.74
 Requires: libseccomp >= 2.3
@@ -55,6 +58,7 @@ BuildRequires: selinux-policy-devel
 BuildRequires: systemd-devel
 BuildRequires: tar
 BuildRequires: which
+BuildRequires: golang
 
 # conflicting packages
 Conflicts: docker
@@ -81,17 +85,20 @@ depending on a particular stack or provider.
 %prep
 %setup -q -c -n src -a 0
 
+# Apply STX patches
+%patch0001 -p1
+
 %build
 
 export DOCKER_GITCOMMIT=%{_gitcommit}
-mkdir -p /go/src/github.com/docker
-ln -s /root/rpmbuild/BUILD/src/engine /go/src/github.com/docker/docker
+mkdir -p ${HOME}/go/src/github.com/docker
+ln -s %{_builddir}/src/engine ${HOME}/go/src/github.com/docker/docker
 
-pushd /root/rpmbuild/BUILD/src/engine
+pushd %{_builddir}/src/engine
 for component in tini "proxy dynamic";do
-    TMP_GOPATH="/go" hack/dockerfile/install/install.sh $component
+    TMP_GOPATH="${HOME}/go" hack/dockerfile/install/install.sh $component
 done
-VERSION=%{_origversion} PRODUCT=docker hack/make.sh dynbinary
+VERSION=%{_origversion} PRODUCT=docker AUTO_GOPATH=1 hack/make.sh dynbinary
 popd
 
 %check
@@ -102,10 +109,10 @@ engine/bundles/dynbinary-daemon/dockerd -v
 install -D -p -m 0755 $(readlink -f engine/bundles/dynbinary-daemon/dockerd) $RPM_BUILD_ROOT/%{_bindir}/dockerd
 
 # install proxy
-install -D -p -m 0755 /usr/local/bin/docker-proxy $RPM_BUILD_ROOT/%{_bindir}/docker-proxy
+install -D -p -m 0755 /tmp/docker-proxy $RPM_BUILD_ROOT/%{_bindir}/docker-proxy
 
 # install tini
-install -D -p -m 755 /usr/local/bin/docker-init $RPM_BUILD_ROOT/%{_bindir}/docker-init
+install -D -p -m 755 /tmp/docker-init $RPM_BUILD_ROOT/%{_bindir}/docker-init
 
 # install systemd scripts
 install -D -m 0644 %{_topdir}/SOURCES/docker.service $RPM_BUILD_ROOT/%{_unitdir}/docker.service
-- 
2.17.1

